### PR TITLE
Add ⭐️Weights & Biases⭐️ Logging

### PR DESCRIPTION
**Pull request by SauravMaheshkar**
This PR aims to add basic [**Weights and Biases**](https://wandb.ai/site) Metric Logging by appending to the existing codebase with minimal changes while supporting Checkpoint uploads as Weights and Biases Artifacts.

Wherever needed, I have used the existing Weights and Biases integrations viz. LightGBM and XGBoost. 

I have validated the performance of all the proposed runs by running 150+ runs, which can be viewed on **[this project page](https://wandb.ai/sauravm/RTDL)** and in detail in an **[accompanying blog post](https://wandb.ai/sauravm/RTDL/reports/Revisiting-Deep-Learning-Models-for-Tabular-Data--VmlldzoxNDE1Njk0)**.

### CODE REVIEW

**Comment by SauravMaheshkar** [comment]
Request for Review @Yura52

**Comment by Yura52** [comment]
@SauravMaheshkar sorry for the slow reply.

Thank you for the report and the pull request!
If I understand correctly, the report contains results for all the models on the California Housing dataset.
Nice to see that they are close to what we report in the paper.

As for the pull request, as of now, we would like to keep the implementation of the paper unchanged and only consider patches that are some kind of fixes. However, we can add a link to the pull request (or to the public fork, if available) in README.
